v2/device.go

```diff
@@ -120,8 +77,48 @@ var newClient = func(opt *mqtt.ClientOptions) mqtt.Client {
 }
 
 func (s *DeviceClient) connect() {
-	s.cli = newClient(s.mqttOpt)
+	p, err := awsiotprotocol.ByURL(s.opt.URL)
+	if err != nil {
+		panic(err)
+	}
+	mqttOpt, err := p.NewClientOptions(
+		&awsiotprotocol.Config{
+			KeyPath:  s.opt.KeyPath,
+			CertPath: s.opt.CertPath,
+			CaPath:   s.opt.CaPath,
+			ClientID: s.opt.ClientID,
+			URL:      s.opt.URL,
+		},
+	)
+	if err != nil {
+		panic(err)
+	}
+
+	s.mqttOpt = mqttOpt
 
+	connectionLost := func(client mqtt.Client, err error) {
+		s.dbg.printf("Connection lost (%s)\n", err.Error())
+		if s.opt.OnConnectionLost != nil {
+			s.opt.OnConnectionLost(s.opt, err)
+		}
+
+		s.stateUpdateCh <- inactive
+	}
+	onConnect := func(client mqtt.Client) {
+		s.dbg.printf("Connection established\n")
+		s.stateUpdateCh <- established
+	}
+
+	s.mqttOpt.OnConnectionLost = connectionLost
+	s.mqttOpt.OnConnect = onConnect
+	if s.opt.Will != nil {
+		s.mqttOpt.SetWill(s.opt.Will.Topic, s.opt.Will.Payload, s.opt.Qos, s.opt.Retain)
+	}
+	s.mqttOpt.SetKeepAlive(s.opt.Keepalive)
+	s.mqttOpt.SetAutoReconnect(false) // MQTT AutoReconnect doesn't work well for mqtts
```

Comment:
Could you update comment?
Manually reconnected to allow updating connection setting on ConnectionLost.